lib/hq/graphql/resource.rb

```diff
@@ -210,13 +199,24 @@ def def_root(field_name, is_array: false, null: true, &block)
               field_name: field_name, resolver: connection_resolver, model_name: model_name
             }
           else
+            resolver = -> {
+              klass = Class.new(::GraphQL::Schema::Resolver) do
+                type = hydrate ? resource.nil_query_object : resource.query_object
+                type type, null: null
+                class_eval(&block) if block
+              end
+
+              constant_name = "#{field_name.to_s.classify}Resolver#{suffix}"
+              resource.send(:remove_const, constant_name) if resource.const_defined?(constant_name, false)
+              resource.const_set(constant_name, klass)
+            }
             ::HQ::GraphQL.root_queries << {
               field_name: field_name, resolver: resolver, model_name: model_name
             }
           end
         end
 
-        def root_query(find_one: true, find_all: true, pagination: true, limit_max: 250)
+        def root_query(find_one: true, find_all: true, hydrate: true, pagination: true, limit_max: 250)
```

Comment:
shouldn't we always hydrate and paginate? I dont' think we need to allow limit_max as a param... just always the same.